### ISSUE TITLE
CI Random Failure in `test_parse_cache_hit`

### ISSUE THREAD

**Issue by kentrutan** [pymoca/pymoca#312]
Had a CI failure last night on my fork where master is up to date with pymoca/pymoca. The log file is attached. Several others in the matrix passed before this failed. I reran the entire CI today and it passed 2nd time around. I have not dug into the details but I thought @jackvreeken might be interested.

[8_test (windows-latest, 3.8).txt](https://github.com/pymoca/pymoca/files/13591627/8_test.windows-latest.3.8.txt)

**Comment by jackvreeken** [comment]
Hmm, that's an interesting one indeed. It oddly enough is able to parse the model twice within the same millisecond, or that's what it looks like. That seems unlikely, so there's probably an assumption on my end about the timing/parallel access that's not right.

The failed assertion leads to the database not being closed properly, which leads to the temporary directory not being able to be cleaned up. That's definitely something I should fix, aside from the timing/modification time issue above.

**Comment by jackvreeken** [comment]
Gave it a go with #313

**Comment by jackvreeken** [comment]
Still not fixed. As far as the "cleaned up" errors are concerned, we can just use `ignore_cleanup_errors ` from Python 3.10 onwards. Currently still supporting 3.8, so in the future maybe. Or fix the tests to really always perfectly clean up after themselves

**Comment by jackvreeken** [comment]
Alright, adding a bit more debug info, I get that (see https://github.com/jackvreeken/pymoca-test-1/actions/runs/7687634083/job/20947890171 )

```
First hit time =  1706466498567864
Updating last_hit for model with hash 'b13ea91d1ac7c6721222e3767aa373739cd575768ebe5e7572cf3b5830fcaabb' (0+untagged.1.g0086d0d)from 1706466498567864 to 1706466498567864
Second hit time in parse() =  1706466498567864
Second hit time =  1706466498567864
1706466498567864 1706466498567864
```

So in other words, `_microseconds_since_epoch()` is returning the same value _twice_. So the issues probably stem from the _time_ resolution on Windows, _not_ something with network storage. So if we update those "last updated" values, we should make sure to always increment by at least _one_ microsecond. That seems like a sane minimum to me. 

I thought I had some 2 second sleep implemented, but apparently that's not triggered. I honestly prefer _not_ sleeping, but I do want to see if it fixes things.

**Comment by jackvreeken** [comment]
Fixed with 2d02861e545246a7afa7dfe7657ea830ed170d91
